sdk/communication/AzureCommunicationChat/Source/NotificationUtils/CryptoUtils.swift

```diff
@@ -142,15 +142,37 @@ func decryptPushNotificationPayload(cipherText: [UInt8], iv: [UInt8], cryptoKey:
     return String(decoding: cryptData, as: UTF8.self)
 }
 
-public struct EncryptionKeyPair {
-    let firstKey: String
-    let secondKey: String
-
-    public init(
-        firstKey: String,
-        secondKey: String
-    ) {
-        self.firstKey = firstKey
-        self.secondKey = secondKey
+internal func generateEncryptionKey() -> String {
+    return SymmetricKey(size: .init(bitCount: 512)).serialize()
+}
+
+// Use the SymmetricKey class in CryptoKit framework to create encryption keys
+extension SymmetricKey {
+    /// Serializes a `SymmetricKey` to a Base64-encoded `String`.
+    func serialize() -> String {
+        return withUnsafeBytes { body in
+            Data(body).base64EncodedString()
+        }
+    }
+}
+
+func splitEncryptionKey(encryptionKey: String) -> [String] {
+    guard var data = Data(base64Encoded: encryptionKey, options: .ignoreUnknownCharacters) else {
+        return ["0000000000000000B00000000000000000000000AES=", "0000000000000000B0000000000000000000000AUTH="]
+    }
```

Comment:
Why would this not be an error? It seems returning an empty key is a recipe for auth bugs.

Reply:
FYI - "something unexpected happens" is based on the fact that the Contoso wants to implement "advanced version of PN" and has already set the PNKeyHandler.

If the Contoso just wants to implement "simple version" and doesn't want to worry about encryption/decryption things, that's a different story. We will not treat this scenario as an error. SDK will register two fake encryption keys for Contoso,

Reply:
Yeah, this is the same rationale as above.